cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -1201,7 +1201,12 @@ def run_query(self, off_line=False, disp_conf=None):
         self.api = api # TODO: we should decide if it's memeber or not
 
         try:
-            self.validate_job_id()
+            # TODO still I am not entirely sure this is a good approach
+            query_status = self.par_dic['query_status']
+            # the idea is that, if the query is new then, there is no analysis_parameters.json file
+            # otherwise, like in the case of the click of the fit button, query_status will be 'ready'
+            # and so an analysis_parameters.json dump is already there
+            self.validate_job_id(query_status != 'new')
```

Comment:
for readability, I would make explicit kw argument here `request_parameters_from_scratch_dir=...`.
but in principle, it does not matter, and also this logic is rather confused and to be revised later.

Reply:
exactly, probably it will work just fine now, but I am not 100% sure regarding its stability